### PR TITLE
Support setting a custom prefix to be used at runtime

### PR DESCRIPTION
**Pull request by timsueberkrueb**
To allow the Liri Text snap application to read the bundled
language specs at runtime, we need to be able to set
different paths for compile time and runtime.
This would be a possible implementation, let me know
if you'd like this to be done differently:
Support `CUSTOM_RUNTIME_DATADIR` and use it for
`ABSOLUTE_LANGUAGE_PATH`.
Rename `DATA_PREFIX` to `DATA_INSTALL_PREFIX` to clarify.

### CODE REVIEW

File: syntax-config.pri [hunk 1]
```diff
@@ -7,11 +7,16 @@ win32|macx {
         DEFINES += GTKSOURCEVIEW_LANGUAGE_PATH=\\\"/usr/share/gtksourceview-3.0/language-specs/\\\"
         message(Using GtkSourceView language specifications.)
     }
+    
+    DATA_INSTALL_PREFIX = $$LIRI_INSTALL_DATADIR/liri-text
+    DATA_RUNTIME_PREFIX = $$CUSTOM_RUNTIME_DATADIR/liri-text
+    isEmpty(DATA_RUNTIME_PREFIX) {
```
Comment (Aldrog) [inline syntax-config.pri]: You're checking DATA_RUNTIME_PREFIX for emptiness right after setting it. Following branch will never be active.

**Review by Aldrog — CHANGES_REQUESTED** [review]

File: syntax-config.pri (hunk 1 above)
Comment (Aldrog) [reply syntax-config.pri]: I think it will be alright if you delete line 12.

Comment (timsueberkrueb) [reply syntax-config.pri]: @Aldrog oh, right, better check if `CUSTOM_RUNTIME_DATADIR` is defined.

Comment (Aldrog) [reply syntax-config.pri]: Why not to simply pass DATA_RUNTIME_PREFIX in special cases like snap packaging?

Comment (timsueberkrueb) [reply syntax-config.pri]: Okay, sure.

**Comment by timsueberkrueb** [comment]
@Aldrog you might want to squash this.

**Comment by timsueberkrueb** [comment]
~~Or, let me do it, don't merge yet.~~
EDIT: okay, here we go.

**Review by Aldrog — APPROVED** [review]

**Comment by timsueberkrueb** [comment]
Thanks @Aldrog!

**Comment by Aldrog** [comment]
Thank you for doing all the snap stuff :)

**Pull request merged**